includes/service-providers/constant_contact/class-newspack-newsletters-constant-contact.php

```diff
@@ -0,0 +1,451 @@
+					[
+						'format'          => 'HTML',
+						'email_addresses' => $emails,
+					]
+				)
+			);
+
+			$data            = $this->retrieve( $post_id );
+			$data['result']  = $result;
+			$data['message'] = sprintf(
+			// translators: Message after successful test email.
+				__( 'Constant Contact test sent successfully to %s.', 'newspack-newsletters' ),
+				implode( ', ', $emails )
+			);
+
+			return \rest_ensure_response( $data );
+
+		} catch ( CtctException $e ) {
+			return $this->manage_ctct_exception( $e );
+		} catch ( Exception $e ) {
+			return new WP_Error(
+				'newspack_newsletters_constant_contact_error',
+				$e->getMessage()
+			);
+		}
+	}
+
+	/**
+	 * Synchronize post with corresponding ESP campaign.
+	 *
+	 * @param WP_POST $post Post to synchronize.
+	 * @return object|null API Response or error.
+	 * @throws Exception Error message.
+	 */
+	public function sync( $post ) {
+		$api_key = $this->api_key();
+		if ( ! $api_key ) {
+			return new WP_Error(
+				'newspack_newsletters_incorrect_post_type',
+				__( 'No Constant Contact API key available.', 'newspack-newsletters' )
+			);
+		}
+		try {
+			$cc             = new ConstantContact( $api_key );
+			$cc_campaign_id = get_post_meta( $post->ID, 'cc_campaign_id', true );
+			$renderer       = new Newspack_Newsletters_Renderer();
+			$content        = $renderer->render_html_email( $post );
+			if ( $cc_campaign_id ) {
+				$campaign = $cc->emailMarketingService->getCampaign( $this->access_token(), $cc_campaign_id ); //phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+
+				$campaign->subject       = $post->post_title;
+				$campaign->email_content = $content;
+
+				$campaign_result = $cc->emailMarketingService->updateCampaign( $this->access_token(), $campaign ); //phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+
+			} else {
+
+				$account_info = $cc->accountService->getAccountInfo( $this->access_token() ); //phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+
+				$initial_sender = __( 'Sender Name', 'newspack-newsletters' );
+				if ( $account_info->organization_name ) {
+					$initial_sender = $account_info->organization_name;
+				} elseif ( $account_info->first_name && $account_info->last_name ) {
+					$initial_sender = $account_info->first_name . ' ' . $account_info->last_name;
+				}
+
+				$verified_email_addresses = $cc->accountService->getVerifiedEmailAddresses( $this->access_token() ); //phpcs:ignore WordPress.NamingConventions.ValidVariableName.UsedPropertyNotSnakeCase
+
+				if ( empty( $verified_email_addresses ) ) {
+					throw new Exception( __( 'There are no verified email addresses in the Constant Contact account.', 'newspack-newsletters' ) );
+				}
+
+				$initial_email_address = $verified_email_addresses[0]->email_address;
+
+				$campaign                       = new Campaign();
+				$campaign->name                 = uniqid();
```

Comment:
Suggestion: prefix that id with e.g. `Newspack Newsletters`, so that it looks less eerie in the CC UI

Reply:
Fixed in https://github.com/Automattic/newspack-newsletters/pull/274/commits/cf5d7c9f292411e1620f2c591f4565b5416d84d8